bugwarrior/config/schema.py

```diff
@@ -172,104 +197,80 @@ def compute_data(cls, values):
     merge_tags: bool = True
     replace_tags: bool = False
-    static_tags: ConfigList = ConfigList([])
-    static_fields: ConfigList = ConfigList(['priority'])
+    static_tags: ConfigList = []
+    static_fields: ConfigList = ["priority"]
 
-    log_level: typing.Literal[
-        ('DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL', 'DISABLED')
-    ] = 'INFO'
+    log_level: Literal["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL", "DISABLED"] = (
+        "INFO"
+    )
     log_file: typing.Optional[LoggingPath] = None
 
 
-class Hooks(pydantic.v1.BaseModel):
-    pre_import: ConfigList = ConfigList([])
+class Hooks(BaseConfig):
+    pre_import: ConfigList = []
 
 
-class Notifications(pydantic.v1.BaseModel):
+class Notifications(BaseConfig):
     notifications: bool = False
-    # Although upstream supports it, pydantic has problems with Literal[None].
-    backend: typing.Optional[
-        typing.Literal[('gobject', 'growlnotify', 'applescript')]
-    ] = None
+    backend: typing.Optional[Literal["gobject", "growlnotify", "applescript"]] = None
     finished_querying_sticky: bool = True
     task_crud_sticky: bool = True
     only_on_new_tasks: bool = False
 
 
-class SchemaBase(pydantic.v1.BaseSettings):
-    class Config(PydanticConfig):
-        # Allow extra top-level sections so all targets don't have to be selected.
-        extra = 'ignore'
+class SchemaBase(BaseConfig):
+    model_config = ConfigDict(extra="ignore")
 
-    hooks: Hooks = Hooks()
-    notifications: Notifications = Notifications()
+    hooks: Hooks = Field(default_factory=Hooks)
+    notifications: Notifications = Field(default_factory=Notifications)
 
 
 class ValidationErrorEnhancedMessages(list):
-    """Methods loosely adapted from pydantic.error_wrappers."""
+    """Pydantic v2 implementation using error.errors()."""
 
-    def __init__(self, error: pydantic.v1.ValidationError):
-        super().__init__(self.flatten(error))
+    def __init__(self, error: pydantic.ValidationError):
+        super().__init__(self._format_error(e) for e in error.errors())
+
+    @staticmethod
+    def _format_error(error: ErrorDetails) -> str:
+        msg = error["msg"]
+        if error["type"] == "extra_forbidden":
+            msg = "unrecognized option"
+        return f"{ValidationErrorEnhancedMessages.display_error_loc(error['loc'])}  <- {msg}\n"
 
     def __str__(self):
-        return '\n'.join(self)
+        return "\n".join(self)
 
     @staticmethod
-    def display_error_loc(e):
-        loc_len = len(e['loc'])
-        if loc_len == 1 or e['loc'][1] == '__root__':
-            return f"[{e['loc'][0]}]"
+    def display_error_loc(loc: tuple) -> str:
+        loc_len = len(loc)
+        if loc_len == 1 or (loc_len > 1 and loc[1] == "__root__"):
+            return f"[{loc[0]}]"
         elif loc_len == 2:
-            # TODO We should be able to display the value itself:
-            # https://github.com/samuelcolvin/pydantic/issues/784
```

Comment:
Per this link, it looks like we should be able to display the value with pydantic v2. So instead of an error message like

```
import_labels_as_tags  <- Input should be a valid boolean, unable to interpret input
```

it could show the whole line:

```
import_labels_as_tags = 100  <- Input should be a valid boolean, unable to interpret input
```